### ISSUE TITLE
pyright 1.1.409 `reportPrivateImportUsage` on `torch.bfloat16` import

### ISSUE THREAD

**Issue by thejcannon** [microsoft/pyright#11405]
**Describe the bug**

Given a file like

```python
from torch import bfloat16
```

Then:

```terminal
me@computer $ uv tool run --with pyright==1.1.408 --with torch==2.8.0 --python 3.12 pyright client.py
0 errors, 0 warnings, 0 informations
me@computer $ uv tool run --with pyright==1.1.409 --with torch==2.8.0 --python 3.12 pyright client.py
/home/josh_cannon/client.py
  /home/josh_cannon/client.py:1:19 - error: "bfloat16" is not exported from module "torch"
    Import from "torch._C" instead (reportPrivateImportUsage)
```

I suspect this is due to https://github.com/microsoft/pyright/pull/11375, but couldn't verify precisely.

This also may just be a bug in `torch` (e.g. `torch` should define a `__init__.pyi` with all the names in it, a subset of which are in `_C/__init__.pyi`)

**Comment by nate-england** [comment]
I'm also curious if there's a good way to silence errors like this targeted to specific usage. Kind of like warning filters in https://docs.pytest.org/en/stable/how-to/capture-warnings.html#controlling-warnings

**Comment by ClaasRostock** [comment]
I can confirm. Same behavior here. Started with update from 1.1.408 to 1.1.409

**Comment by nileracecrew** [comment]
Seeing what I think is a related issue in Bokeh after updating to  pyright 1.1.409. Both [torch](https://github.com/pytorch/pytorch/blob/4e3940c4a48e1e2b74e3cd3719a4015a1d6db7d8/torch/__init__.py#L2217) and [Bokeh](https://github.com/bokeh/bokeh/blob/54c75abcae66611a4e53c64e36e3e0cc94067966/src/bokeh/models/widgets/__init__.py#L48) generate `__all__` dynamically. I'm not sure if pyright is expected to expand these, but the imports flagged as private are populated in `__all__` eventually (at least in Bokeh).

**Comment by renzph** [comment]
I'm getting the same behaviour. Could it also be due to https://github.com/microsoft/pyright/pull/11237 ?

**Comment by cosinequanon** [comment]
This looks fixed now:
```
$ uv tool run --with pyright==1.1.411 --with torch==2.8.0 --python 3.12 pyright client.py
Installed 12 packages in 307ms
0 errors, 0 warnings, 0 informations
```

**Comment by rchiodo** [comment]
Thanks for verifying.
